lib/src/native/realm_core.dart

```diff
@@ -190,7 +190,9 @@ class _RealmCore {
         final syncConfigPtr = _realmLib.invokeGetPointer(() => _realmLib.realm_flx_sync_config_new(config.user.handle._pointer));
         try {
           _realmLib.realm_sync_config_set_session_stop_policy(syncConfigPtr, config.sessionStopPolicy.index);
-          _realmLib.realm_config_set_sync_config(configPtr, syncConfigPtr);
+          _realmLib.realm_sync_config_set_error_handler(
+              syncConfigPtr, Pointer.fromFunction(_syncErrorHandlerCallback), config.toPersistentHandle(), nullptr);
```

Comment:
I believe this should be a weak handle to the handler directly. Why try to persist the config?

Reply:
I made it persistent, because if it is released we will not be able to receive the callback. This config is created when the app is created and it is expected to be used during the whole lifecycle of the Session.